### ISSUE TITLE
Hide 'View Docs' button if no default version

### ISSUE THREAD

**Issue by agjohnson** [rtfd/readthedocs.org#3398]
If a project has no `latest` or `stable` version enabled, we shouldn't show this button. Otherwise, the link still goes to the `latest` version, which 404s.

Design decision needed:

- Is there something we should point to?
- Is there a better UX here? Maybe pointing to the versions page instead of just removing the view docs button?

**Comment by stsewd** [comment]
I just submitted a PR (#3446) to solve this part:

> If we're on a build, use the build version for the view docs link! This gets me every time 

I think the `View Docs` button must remain the same on all the _project area_, otherwise it would be a little confusing.

Maybe we can display a "gray" `View Docs` button when there is any versions enabled.

**Comment by humitos** [comment]
I've been using the beta templates for a while now (https://beta.readthedocs.org/) and this button is something that I missed because it's hard to have "quick access to the documentation of this project". In particular when there a few active versions.

When you go to a project details' page all the versions are listed. You have to visually find the `⭐ Default` label on the version and then go to the 📖 icon at the right side of the row. This is a slow process and gets slower when the project have many versions.

I'd like to keep this button in the new templates using a big button saying something like "View documentation for default version" or similar. cc @agjohnson

**Comment by agjohnson** [comment]
Yeah it would be fine to add this button back, I've left space for a button like this on the bottom of the project sub-navigation block:

![image](https://github.com/readthedocs/readthedocs.org/assets/1140183/6fd69524-0185-4903-91a7-5c78082c4268)

There is already a button on the build detail page in the same place:
![image](https://github.com/readthedocs/readthedocs.org/assets/1140183/b5b07c9c-1253-493b-b743-8a720e718fc2)

However, currently the template logic is this:

https://github.com/readthedocs/readthedocs.org/blob/fa54900abc2f1f989e5a608a1c0b75905f075719/readthedocs/templates/core/project_bar_base.html#L9C1-L14C16

Instead, the template probably needs the project's default Version instance passed from the view. The above template logic can't inspect attributes about the default version, if it is active (the button is enabled) or the version name directly (add the text/popup "View docs for the 3.0 docs" etc). I say pass it in from the view because I've been removing query logic from templates, and prefer to keep that in application logic where it can be optimized.

**Comment by humitos** [comment]
> Instead, the template probably needs the project's default Version instance passed from the view. The above template logic can't inspect attributes about the default version, if it is active (the button is enabled) or the version name directly (add the text/popup "View docs for the 3.0 docs" etc).

I would implement "no logic" at all here and keep it simple. I'd _always_ show that link to the default domain for this project `https://<slug>.readthedocs.io` and leave that URL to do the magic to show the default version the project has configured.

I don't think the reader needs the "3.0" on the link and I don't think we should implement any logic to "show/hide" the link depending on that logic. This may be more confusing than helpful. Readers wanting to read a specific version can click on the "book" icon from the particular version they want to read. However, most of the users would just want to "go to the documentation version _the author wants me to read_ of this project".

Also, showing the "default version" on the link would bring some issues with version with long names like `release/september/3.2.11` or similar things.

**Comment by agjohnson** [comment]
> Readers wanting to read a specific version can click on the "book" icon from the particular version they want to read. 

I will say that I prefer this UX entirely, over any version of a top "View docs" button, as the version list has addressed these issues already.

Originally, I wanted a version listing that put the default/latest at the top of the version list, but for now the default sorting is by latest built. I think a default sorting that surfaces default/latest/stable first solves the problem you are talking about in a better way, as it keeps all of the interactions to view documentation in a single place instead of splitting the UX up into competing patterns.

If the default/latest/stable versions were ordered first in the version list, would you still need a "View docs" button? Is it worth trying this again first?

I'm also not against the view docs button, but I intended the version listing to replace the need for this button for many of the reasons here.

> I'd always show that link to the default domain for this project https://<slug>.readthedocs.io

Is there significance to the default domain here?

I don't think it changes anything, but this should be the *custom domain* if one is defined, not just the default domain. But this should be the logic from the URL resolver already though.

> I don't think the reader needs the "3.0" on the link

Agreed it doesn't need to be button text. I'd add the version as some extra UI element near the button to indicate what we're linking to.

This was the intention behind putting the links in the version listing too though, to make it very explicit what you were about to view. The "View docs" button is not clear about what it is linking to otherwise, and this is confusing right now. 

> I don't think we should implement any logic to "show/hide" the link depending on that logic.

The current buttons for view docs do already have an enabled/disabled state, both the version listing buttons and the build detail "View docs" button. I'd want to retain this. There are a few cases where "View docs" will lead the user immediately to a 404/403 error page: when there is no output for the default version (new projects and projects that change their default version) and when the user doesn't have access to the default version. Perhaps there are more too.

For this, we just need access to the version instance of the project's default version though, it's not a major addition of logic.

> This may be more confusing than helpful. 

Yeah this could be confusing if we just disabled the button without any extra information. I think we can identify what makes it confusing UX and work that into the plan though. I don't want to *only* disable the button, but the button would also have some sort of supporting UI/icon/popup to describe *why* the button is disabled.

> Also, showing the "default version" on the link would bring some issues with version with long names like release/september/3.2.11 or similar things.

Yeah definitely, I also wouldn't want the button text to include the version name for this reason. But it should have text/popup the does describe the version we're linking to.
